internal/services/pam/pam_test.go

```diff
@@ -0,0 +1,606 @@
+
+	gotResp, _ := client.GetPreviousBroker(context.Background(), &authd.GPBRequest{Username: "user"})
+	require.Equal(t, "local", gotResp.GetPreviousBroker(), "GetPreviousBroker did not return the correct broker")
+
+	gotResp, _ = client.GetPreviousBroker(context.Background(), &authd.GPBRequest{Username: "does not have a broker"})
+	require.Empty(t, gotResp.GetPreviousBroker(), "GetPreviousBroker should return nil when the user has no broker assigned")
+}
+
+func TestSelectBroker(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		mockResponse string
+		brokerID     string
+		username     string
+
+		wantErr bool
+	}{
+		"Successfully select a broker and creates the session": {username: "success"},
+
+		"Error when username is empty":                    {wantErr: true},
+		"Error when brokerID is empty":                    {username: "empty broker", brokerID: "empty", wantErr: true},
+		"Error when broker does not exist":                {username: "no broker", brokerID: "does not exist", wantErr: true},
+		"Error when broker does not provide a session ID": {username: "NS_no_id", wantErr: true},
+		"Error when starting the session":                 {username: "NS_error", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			brokerCfgDir := t.TempDir()
+			brokerCfg := testutils.StartBusBrokerMock(t, brokerCfgDir)
+
+			client := startClient(t, brokerCfgDir)
+
+			if tc.brokerID == "" {
+				tc.brokerID = getGeneratedID(t, client, filepath.Base(brokerCfg))
+			} else if tc.brokerID == "empty" {
+				tc.brokerID = ""
+			}
+
+			sbRequest := &authd.SBRequest{
+				BrokerId: tc.brokerID,
+				Username: tc.username,
+			}
+			sbResp, err := client.SelectBroker(context.Background(), sbRequest)
+			if tc.wantErr {
+				require.Error(t, err, "SelectBroker should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "SelectBroker should not return an error, but did")
+			got := fmt.Sprintf("ID: %s\nEncryption Key: %s\n",
+				strings.ReplaceAll(sbResp.GetSessionId(), tc.brokerID, "BROKER_ID"),
+				sbResp.GetEncryptionKey())
+			want := testutils.LoadWithUpdateFromGolden(t, got)
+			require.Equal(t, want, got, "SelectBroker returned an unexpected response")
+		})
+	}
+}
+
+func TestGetAuthenticationModes(t *testing.T) {
+	t.Parallel()
+
+	required, optional := "required", "optional"
+	supportedEntries := "optional:chars,chars_password"
+	optionalWithBooleans := "optional:true,false"
+
+	normalLayout := &authd.UILayout{
+		Type:    "form",
+		Label:   &required,
+		Button:  &optional,
+		Wait:    &optionalWithBooleans,
+		Entry:   &supportedEntries,
+		Content: &optional,
+	}
+	invalidLayout := &authd.UILayout{
+		Type: "",
+	}
+
+	tests := map[string]struct {
+		brokerID  string
+		sessionID string
+		username  string
+		layout    *authd.UILayout
+
+		wantErr bool
+	}{
+		"Successfully get authentication modes": {},
```

Comment:
Can we get more than one authentication modes as it’s the general use case from the mock? (I don’t think we need 2 test cases, but this will show in the golden file that we have a list here)